tests/transports/test_base.py

```diff
@@ -77,88 +82,85 @@ def test_flush_time(mock_send, caplog):
     transport._last_flush = timeit.default_timer() - 5.1
     with caplog.at_level("DEBUG", "elasticapm.transport"):
         transport.queue("error", {})
+        transport.close()
     record = caplog.records[0]
     assert "5.1" in record.message
-    assert mock_send.call_count == 1
+    assert mock_send.call_count == 2  # one for the send, one for the close
     assert transport._queued_data is None
 
 
-@mock.patch("elasticapm.transport.base.Transport.send")
-def test_flush_time_size(mock_send, caplog):
+@mock.patch("elasticapm.transport.base.Transport.flush")
+def test_flush_time_size(mock_flush, caplog):
+    transport = Transport(metadata={}, max_buffer_size=100)
     with caplog.at_level("DEBUG", "elasticapm.transport"):
-        transport = Transport(metadata={}, max_buffer_size=100)
         # we need to add lots of uncompressible data to fill up the gzip-internal buffer
         for i in range(9):
             transport.queue("error", "".join(random.choice(string.ascii_letters) for i in range(2000)))
-    record = caplog.records[1]
-    assert "queue size" in record.message
-    assert mock_send.call_count == 1
-    assert transport._queued_data is None
+    transport.close()
+    assert mock_flush.call_count == 2
 
 
 @mock.patch("elasticapm.transport.base.Transport.send")
 def test_forced_flush(mock_send, caplog):
+    transport = Transport(metadata={}, max_buffer_size=1000, compress_level=0)
     with caplog.at_level("DEBUG", "elasticapm.transport"):
-        transport = Transport(metadata={}, max_buffer_size=1000, compress_level=0)
         transport.queue("error", "x", flush=True)
-    record = caplog.records[0]
-    assert "forced" in record.message
```

Comment:
Unfortunately, it seems the caplog fixture has some difficulties in multithreaded environments (or I have difficulties using it correctly), so I had to remove some of these asserts